components/Header.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  showFavoritesBtnForRoutes,
+  showSearchBarForRoutes,
+  showViewAllBtnForRoutes,
+} from "@/constants/const";
+import { useAuth } from "@/context/AuthContext";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React, { useState } from "react";
+import Image from "next/image";
+import { logOut } from "@/store/firebaseHelpers";
+
+const SearchBar = ({
+  setGlobalQuery,
+}: {
+  setGlobalQuery: (query: string) => void;
+}) => {
+  const [searchQuery, setSearchQuery] = useState("");
+  return (
+    <div>
+      <form>
+        <div className="max-w-xl">
+          <div className="flex">
+            <div className="flex rounded-md overflow-hidden">
+              <input
+                type="text"
+                className="px-5 w-full rounded-md rounded-r-none text-neutral-900 focus:outline-none"
+                onChange={(e) => {
+                  setSearchQuery(e.target.value);
+                  setGlobalQuery(e.target.value);
+                }}
+                value={searchQuery}
+              />
+              <button className="bg-neutral-900  px-6 text-lg py-3 rounded-r-md font-semibold">
+                <svg
+                  xmlns="http://www.w3.org/2000/svg"
+                  className="w-5 h-5"
+                  fill="none"
+                  viewBox="0 0 24 24"
+                  stroke="currentColor"
+                  strokeWidth={2}
+                >
+                  <path
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
+                  />
+                </svg>
+              </button>
+            </div>
+          </div>
+        </div>
+      </form>
+    </div>
+  );
+};
+
+const Header = () => {
+  const { setSearchQuery } = useAuth();
+  const router = useRouter();
+  return (
+    <div className=" text-white py-10 px-20 flex justify-between items-center sticky top-0 bg-white bg-opacity-20">
+      <div className="flex items-center">
+        <span className="text-3xl font-bold mr-2">PokeDex</span>
+        <Image
+          height={20}
+          width={40}
+          src={"/pokeball-png-45330.png"}
+          alt={""}
+        />
+      </div>
+      <>
+        {showSearchBarForRoutes?.includes(router.pathname) ? (
```

Comment:
Too much inline logic here
Try something like a `renderHeaderContent` function for ease